src/images.js

```diff
@@ -0,0 +1,139 @@
+'use strict'
+
+const proc = require('./proc')
+const output = require('./output')
+const cp = require('child_process')
+const fs = require('fs')
+const path = require('path')
+const crypto = require('crypto')
+
+const images = {
+  /**
+   * Builds a docker image, naming it according to the parent folder and tagging it
+   * with "bc_" followed by a hash of the Dockerfile used to build it.
+   * @param dockerfile
+   * @param imageName
+   * @returns {Promise.<T>}
+   */
+  buildImage: (dockerfile, imageName) => {
+    output.info(`Building image from ${dockerfile}`)
+    output.line()
+    return proc.run([
+      'build',
+      '-f', path.resolve(process.cwd(), dockerfile),
+      '-t', imageName,
+      process.cwd()
+    ]).then(() => {
+      output.line()
+      output.success('Image built successfully!')
+      return imageName
+    }).catch(e => {
+      output.line()
+      output.error('Build failed')
+      throw e
+    })
+  },
+  /**
+   * Deletes a docker image
+   * @param {Promise} imageName The name of the image to be deleted in the format name:tag
+   */
+  deleteImage: (imageName) => Promise.resolve().then(() => {
+    const delSpinner = output.spinner(`Deleting old image: ${imageName}`)
+    const cmd = `docker rmi ${imageName}`
```

Comment:
`-f` in here may prevent some issues...

Reply:
I played with this when I was using the image ID to do deletes -- the issue with -f is that it forces a delete even when it's an intermediate of another container. It's not the smoothest move in the world for someone to base a new container on top of a binci build, but if someone did, throwing in `-f` would delete their work. That's why I went with the only-delete-the-most-recent model -- if one fails to delete (because someone referenced it elsewhere) then that will be the only time it fails.

Reply:
Since you'll be working with this feature most directly I assume you'll catch if there is a benefit in the `-f` flag down the road (just assuming no one would be strangely building off binci-gen'd images). I just remember how stubborn Docker is on some environments (*cough* *cough* OSX) with things like `stop` and `rm`.

Reply:
hahaha, thankfully I've not seem the same hangup on the images side compared to the container side. It just seems to bitch (Without the -f) if you're trying to delete an image with dependents, or if you're trying to delete an ID that has multiple tags.